projects/admin-core/src/lib/useradmin/services/userdetails.service.ts

```diff
@@ -0,0 +1,109 @@
+import { BehaviorSubject, catchError, Observable, of, Subject } from 'rxjs';
+import { Inject, Injectable, OnDestroy } from '@angular/core';
+import { TAILORMAP_ADMIN_API_V1_SERVICE, TailormapAdminApiV1ServiceModel, UserModel } from '@tailormap-admin/admin-api';
+import { SnackBarMessageComponent } from '@tailormap-viewer/shared';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserDetailsService implements OnDestroy {
+  // @ts-ignore
+  private selectedUser: BehaviorSubject<UserModel> = new BehaviorSubject<UserModel>(null);
+  public selectedUser$: Observable<UserModel> = this.selectedUser.asObservable();
+  private userList: BehaviorSubject<UserModel[]> = new BehaviorSubject<UserModel[]>([]);
+  public userList$: Observable<UserModel[]> = this.userList.asObservable();
+  private destroyed = new Subject();
+
+  public constructor(@Inject(TAILORMAP_ADMIN_API_V1_SERVICE) private adminApiService: TailormapAdminApiV1ServiceModel,
+                     private snackBar: MatSnackBar) {
+    this.getUsers();
+  }
+
+  public ngOnDestroy() {
+    this.destroyed.next(null);
+    this.destroyed.complete();
+  }
+
+  public getUsers() {
+    this.adminApiService.getUsers$()
+      .pipe(
+        catchError(err => {
+          this.showErrorMessage($localize`Error while getting users ${err}`);
+          return of(null);
+        }),
+      )
+      .subscribe(users => {
+        if (users) {
+          this.userList.next(users);
+        }
+      });
+  }
+
+  public selectUser(username: string): void {
+    this.adminApiService.getUser$(username, 'GroupName')
+      .pipe(
+        catchError(() => {
+          this.showErrorMessage($localize`Error while getting user ${username}`);
+          return of(null);
+        }))
+      .subscribe(user => {
+        if (user) {
+          if (user.validUntil) {
```

Comment:
Maybe its better to move this to the API service? Since you already modify the user object there for the groups.

Reply:
OK